File: 110-balanced-binary-tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isBalanced(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: bool
        """
        isbalanced=[True]
        def height(root):
            if root is None:
                return 0

            left=height(root.left)
            right=height(root.right)
            
            if abs(left-right)>1:
                isbalanced[0]=False
                return 0
            
            return max(left,right)+1
        
        height(root)
        return isbalanced[0]

File: 110-balanced-binary-tree/balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool balanced=true;
    int helper(TreeNode* root){
        if(root==nullptr) return 0;
        int leftHeight=this->helper(root->left);
        int rightHeight=this->helper(root->right);
        if(abs(leftHeight-rightHeight)>1){
            this->balanced=false;
            return 0;
        }
        return max(leftHeight,rightHeight)+1;
    }
    bool isBalanced(TreeNode* root) {
        helper(root);
        return this->balanced;
    }
};